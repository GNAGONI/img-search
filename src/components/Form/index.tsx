import { SyntheticEvent } from 'react';
import { styled } from '@mui/material/styles';

type Props = {
  onSubmit: (e: SyntheticEvent) => void;
  children: React.ReactNode;
};

const CustomStyledForm = styled('form')({
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'center',
  alignItems: 'center',
  padding: '16px',
})


const Form = (props: Props) => {
  return (
    <CustomStyledForm onSubmit={props.onSubmit}>
      {props.children}
    </CustomStyledForm>
  );
};

export default Form;
