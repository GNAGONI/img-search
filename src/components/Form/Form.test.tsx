import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Form from './index';

describe('Form', () => {
  it('should have snapshot', () => {
    const mockCallback = jest.fn();
    const tree = renderer
      .create(
        <Form onSubmit={mockCallback}>
          <></>
        </Form>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should behave correctly', () => {
    const mockCallback = jest.fn();
    const { container } = render(
      <Form onSubmit={mockCallback}>
        <input type="text" />
        <button>Test</button>
      </Form>
    );

    const form = container.querySelector('form');
    expect(form).toBeInTheDocument();
    expect(form).toBeVisible();

    const button = screen.getByRole('button', { name: 'Test' });
    expect(button).toBeInTheDocument();
    expect(button).toBeVisible();

    const input = screen.getByRole('textbox');
    expect(input).toBeInTheDocument();
    expect(input).toBeVisible();

    if (form) {
      fireEvent.submit(form);
      expect(mockCallback).toHaveBeenCalledTimes(1);
    }
  });
});
