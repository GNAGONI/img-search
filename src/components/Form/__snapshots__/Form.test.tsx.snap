// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should have snapshot 1`] = `
<form
  className="css-epn0vg"
  onSubmit={[MockFunction]}
/>
`;
